src/world/area_hos/hos_05/A2AAC0.c

```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
 }
 
 // adjusts properties of EmitterVolume:GoldShimmer2 effect
-INCLUDE_ASM(s32, "world/area_hos/hos_05/A2AAC0", func_80240E50_A2B090);
+ApiStatus func_80240E50_A2B090(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 posX;
+    s32 posY;
+    f32 posZ;
+    EffectInstance* effect;
+
+    effect = (EffectInstance*) evt_get_variable(script, *args++);
+    evt_get_variable(script, *args++);
+    posX = evt_get_float_variable(script, *args++);
+    posY = evt_get_float_variable(script, *args++);
+    posZ = evt_get_float_variable(script, *args++);
+    effect->data.miscParticles->pos.x = (f32) posX;
```

Comment:
no casts needed (including the double cast on the last one)

Reply:
also probably can move these inline with the var declarations as well (?) unless we don't want to include the useless var